source/jormungandr/jormungandr/interfaces/v1/Uri.py

```diff
@@ -175,7 +175,9 @@ def get(self, region=None, lon=None, lat=None, uri=None, id=None):
                 if not region:
                     abort(404, message="Unable to find an object for the uri %s" % args["external_code"])
             else:
-                abort(503, message="Not implemented yet")
+                id = i_manager.instances[region].has_external_code(type_, args["external_code"])
+                if id == None:
+                    abort(404, message="Unable to find an object for the uri %s" % args["external_code"])
```

Comment:
Same above (sorry it becomes long, and it was there before your changes) : `uri`==`id`!=`external_code`
```suggestion
                    abort(404, message="Unable to find an object for the external_code %s" % args["external_code"])
```

Reply:
hoooo that was the message for no coverage case... 
fix it fix it :construction_worker_man: